network/common/src/main/java/org/apache/spark/network/util/JavaUtils.java

```diff
@@ -121,4 +123,69 @@ private static boolean isSymlink(File file) throws IOException {
     }
     return !fileInCanonicalDir.getCanonicalFile().equals(fileInCanonicalDir.getAbsoluteFile());
   }
+
+  private static ImmutableMap<String, TimeUnit> timeSuffixes = 
+    ImmutableMap.<String, TimeUnit>builder()
+      .put("us", TimeUnit.MICROSECONDS)
+      .put("ms", TimeUnit.MILLISECONDS)
+      .put("s", TimeUnit.SECONDS)
+      .put("min", TimeUnit.MINUTES)
+      .put("h", TimeUnit.HOURS)
+      .put("d", TimeUnit.DAYS)
+      .build();
+
+  /**
+   * Convert a passed time string (e.g. 50s, 100ms, or 250us) to a time count for
+   * internal use. If no suffix is provided a direct conversion is attempted.
+   */
+  private static long parseTimeString(String str, TimeUnit unit) throws NumberFormatException {
+    String timeError = "Time must be specified as seconds (s), " +
```

Comment:
This can just be part of the `catch` block and/or a direct part of the exception constructor? I was surprised to see the error message it as the first statement in the method.